### ISSUE TITLE
lxc-start hangs for unprivileged container

### ISSUE THREAD

**Issue by asavonic** [lxc/lxc#2252]
# Required information

 * Distribution: void-linux
 * Distribution version: rolling
 * The output of
   * `lxc-start --version`
      `2.1.1`
   * `lxc-checkconfig`
      ```
      --- Namespaces ---
      Namespaces: enabled
      Utsname namespace: enabled
      Ipc namespace: enabled
      Pid namespace: enabled
      User namespace: enabled
      Network namespace: enabled

      --- Control groups ---
      Cgroups: enabled

      Cgroup v1 mount points: 
      /sys/fs/cgroup/cpuset
      /sys/fs/cgroup/cpu
      /sys/fs/cgroup/cpuacct
      /sys/fs/cgroup/blkio
      /sys/fs/cgroup/memory
      /sys/fs/cgroup/devices
      /sys/fs/cgroup/freezer
      /sys/fs/cgroup/net_cls
      /sys/fs/cgroup/perf_event
      /sys/fs/cgroup/net_prio
      /sys/fs/cgroup/hugetlb
      /sys/fs/cgroup/pids

      Cgroup v2 mount points: 


      Cgroup v1 systemd controller: /usr/bin/lxc-checkconfig: 156: /usr/bin/lxc-checkconfig: printf \033[1;31m: not found

      Cgroup v1 clone_children flag: enabled
      Cgroup device: enabled
      Cgroup sched: enabled
      Cgroup cpu account: enabled
      Cgroup memory controller: enabled
      Cgroup cpuset: enabled

      --- Misc ---
      Veth pair device: enabled, not loaded
      Macvlan: enabled, not loaded
      Vlan: enabled, loaded
      Bridges: enabled, not loaded
      Advanced netfilter: enabled, not loaded
      CONFIG_NF_NAT_IPV4: enabled, not loaded
      CONFIG_NF_NAT_IPV6: enabled, not loaded
      CONFIG_IP_NF_TARGET_MASQUERADE: enabled, not loaded
      CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
      CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, not loadedCONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
      FUSE (for use with lxcfs): enabled, loaded

      --- Checkpoint/Restore ---
      checkpoint restore: enabled
      CONFIG_FHANDLE: enabled
      CONFIG_EVENTFD: enabled
      CONFIG_EPOLL: enabled
      CONFIG_UNIX_DIAG: enabled
      CONFIG_INET_DIAG: enabled
      CONFIG_PACKET_DIAG: enabled
      CONFIG_NETLINK_DIAG: enabled
      File capabilities: enabled

      Note : Before booting a new kernel, you can check its configuration
      usage : CONFIG=/path/to/config /usr/bin/lxc-checkconfig
      ```
   * `uname -a`
     `Linux desktop 4.15.12_1 #1 SMP PREEMPT Thu Mar 22 13:07:32 UTC 2018 x86_64 GNU/Linux`
   * `cat /proc/self/cgroup`
     ```
     13:name=systemd:/asavonic
     12:pids:/asavonic
     11:hugetlb:/asavonic
     10:net_prio:/asavonic
     9:perf_event:/asavonic
     8:net_cls:/asavonic
     7:freezer:/asavonic
     6:devices:/asavonic
     5:memory:/asavonic
     4:blkio:/asavonic
     3:cpuacct:/asavonic
     2:cpu:/asavonic
     1:cpuset:/asavonic     
     ```
   * `cat /proc/1/mounts`
     ```
     proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
     sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
     devtmpfs /dev devtmpfs rw,nosuid,noexec,size=8163460k,nr_inodes=2040865,mode=755 0 0
     devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
     tmpfs /dev/shm tmpfs rw,nosuid,nodev,noexec 0 0
     tmpfs /run tmpfs rw,nosuid,nodev,noexec,mode=755 0 0
     /dev/sda2 / ext4 rw,relatime,data=ordered 0 0
     securityfs /sys/kernel/security securityfs rw,relatime 0 0
     cgroup /sys/fs/cgroup tmpfs rw,relatime,mode=755 0 0
     cgroup /sys/fs/cgroup/cpuset cgroup rw,relatime,cpuset,clone_children 0 0
     cgroup /sys/fs/cgroup/cpu cgroup rw,relatime,cpu 0 0
     cgroup /sys/fs/cgroup/cpuacct cgroup rw,relatime,cpuacct 0 0
     cgroup /sys/fs/cgroup/blkio cgroup rw,relatime,blkio 0 0
     cgroup /sys/fs/cgroup/memory cgroup rw,relatime,memory 0 0
     cgroup /sys/fs/cgroup/devices cgroup rw,relatime,devices 0 0
     cgroup /sys/fs/cgroup/freezer cgroup rw,relatime,freezer 0 0
     cgroup /sys/fs/cgroup/net_cls cgroup rw,relatime,net_cls 0 0
     cgroup /sys/fs/cgroup/perf_event cgroup rw,relatime,perf_event,release_agent=/run/cgmanager/agents/cgm-release-agent.perf_event 0 0
     cgroup /sys/fs/cgroup/net_prio cgroup rw,relatime,net_prio 0 0
     cgroup /sys/fs/cgroup/hugetlb cgroup rw,relatime,hugetlb,release_agent=/run/cgmanager/agents/cgm-release-agent.hugetlb 0 0
     cgroup /sys/fs/cgroup/pids cgroup rw,relatime,pids,release_agent=/run/cgmanager/agents/cgm-release-agent.pids 0 0
     /dev/sda3 /home ext4 rw,relatime,data=ordered 0 0
     /dev/sdb6 /data ext4 rw,relatime,stripe=32740,data=ordered 0 0
     tmpfs /tmp tmpfs rw,nosuid,nodev,relatime 0 0
     ```

# Issue description

I'm trying to setup unprivileged container, but cannot get past lxc-create:
lxc-start hangs without any output. All other lxc commands start hanging after that too, until I kill lxc monitor processes:
```
asavonic  2176  0.0  0.0      0     0 pts/0    Z+   17:28   0:00 [lxc-start] <defunct>
asavonic  2177  0.0  0.0  56908  4644 ?        Ss   17:28   0:00 [lxc monitor] /home/asavonic/.local/share/lxc p3
asavonic  2184  0.0  0.0  56908   736 ?        S    17:28   0:00 [lxc monitor] /home/asavonic/.local/share/lxc p3
asavonic  2189  0.0  0.0  56908   744 ?        S    17:28   0:00 [lxc monitor] /home/asavonic/.local/share/lxc p3
```

# Steps to reproduce

 1. `sudo cgm create all $USER`
 2. `sudo cgm chown all $USER $(id -u) $(id -g)`
 3. `cgm movepid all $USER $$`
 4. `sudo sv status dbus`
    `run: dbus: (pid 732) 3379s`
 5. `sudo sv status cgmanager`
    `run: cgmanager: (pid 727) 3416s`
 6. `lxc-create -t download -n p3`
     ```
     [...]
     Distribution: 
     ubuntu
     Release: 
     xenial
     Architecture: 
     amd64
     ```
 7. `lxc-start -n p3 --logfile p3.log --logpriority TRACE`
    <hangs with no output>

# Information to attach

 - [X] container log
   ```
      lxc-start p3 20180401142858.595 INFO     lxc_start_ui - tools/lxc_start.c:main:280 - using rcfile /home/asavonic/.local/share/lxc/p3/config
      lxc-start p3 20180401142858.595 INFO     lxc_confile - confile.c:set_config_idmaps:1556 - Read uid map: type u nsid 0 hostid 100000 range 65536
      lxc-start p3 20180401142858.595 INFO     lxc_confile - confile.c:set_config_idmaps:1556 - Read uid map: type g nsid 0 hostid 100000 range 65536
      lxc-start p3 20180401142858.595 TRACE    lxc_commands - commands.c:lxc_cmd:290 - command get_init_pid tries to connect command socket
      lxc-start p3 20180401142858.595 TRACE    lxc_commands - commands.c:lxc_cmd:295 - command get_init_pid failed to connect command socket: Connection refused
      lxc-start p3 20180401142858.595 TRACE    lxc_commands - commands.c:lxc_cmd:290 - command get_init_pid tries to connect command socket
      lxc-start p3 20180401142858.595 TRACE    lxc_commands - commands.c:lxc_cmd:295 - command get_init_pid failed to connect command socket: Connection refused
      lxc-start p3 20180401142858.596 WARN     lxc_cgmanager - cgroups/cgmanager.c:cgm_get:993 - do_cgm_get exited with error
      lxc-start p3 20180401142858.596 TRACE    lxc_commands - commands.c:lxc_cmd:290 - command get_state tries to connect command socket
      lxc-start p3 20180401142858.596 TRACE    lxc_commands - commands.c:lxc_cmd:295 - command get_state failed to connect command socket: Connection refused
      lxc-start p3 20180401142858.596 TRACE    lxc_start - start.c:lxc_init_handler:583 - Created anonymous pair {5,6} of unix sockets
      lxc-start p3 20180401142858.596 TRACE    lxc_start - start.c:lxc_init_handler:592 - unix domain socket 7 for command server is ready
      lxc-start p3 20180401142858.596 INFO     lxc_container - lxccontainer.c:do_lxcapi_start:883 - Attempting to set proc title to [lxc monitor] /home/asavonic/.local/share/lxc p3
      lxc-start p3 20180401142858.596 INFO     lxc_start - start.c:lxc_check_inherited:257 - Closed inherited fd: 3.
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:607 - initialized LSM
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .reject_force_umount  # comment this to allow umount -f;  not recommended.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for reject_force_umount action 0(kill).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:276 - Setting Seccomp rule to reject force umounts.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for reject_force_umount action 0(kill).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:276 - Setting Seccomp rule to reject force umounts.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:276 - Setting Seccomp rule to reject force umounts.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .[all].
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .kexec_load errno 1.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for kexec_load action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for kexec_load action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .open_by_handle_at errno 1.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for open_by_handle_at action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for open_by_handle_at action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .init_module errno 1.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for init_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for init_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .finit_module errno 1.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for finit_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for finit_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:435 - processing: .delete_module errno 1.
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:610 - Adding native rule for delete_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:614 - Adding compat rule for delete_module action 327681(errno).
      lxc-start p3 20180401142858.597 INFO     lxc_seccomp - seccomp.c:parse_config_v2:624 - Merging in the compat Seccomp ctx into the main one.
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:613 - read seccomp policy
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_serve_state_clients:373 - set container state to STARTING
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_serve_state_clients:376 - no state clients registered
      lxc-start p3 20180401142858.597 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
      lxc-start p3 20180401142858.597 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:620 - set container state to "STARTING"
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:648 - set environment variables
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:654 - ran pre-start hooks
      lxc-start p3 20180401142858.597 DEBUG    lxc_start - start.c:setup_signal_fd:301 - Set SIGCHLD handler with file descriptor: 3.
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:665 - set up signal fd
      lxc-start p3 20180401142858.597 DEBUG    console - console.c:lxc_console_peer_default:450 - process does not have a controlling terminal
      lxc-start p3 20180401142858.597 TRACE    lxc_start - start.c:lxc_init:672 - created console
      lxc-start p3 20180401142858.597 DEBUG    lxc_conf - conf.c:chown_mapped_root:2891 - trying to chown "/dev/pts/1" to 1000
      lxc-start p3 20180401142858.628 TRACE    lxc_conf - conf.c:lxc_ttys_shift_ids:2969 - chowned console "/dev/pts/1"
      lxc-start p3 20180401142858.628 TRACE    lxc_start - start.c:lxc_init:678 - shifted tty ids
      lxc-start p3 20180401142858.628 INFO     lxc_start - start.c:lxc_init:680 - container "p3" is initialized
      lxc-start p3 20180401142858.629 INFO     lxc_cgroup - cgroups/cgroup.c:cgroup_init:67 - cgroup driver cgmanager initing for p3
      lxc-start p3 20180401142858.633 INFO     lxc_start - start.c:lxc_spawn:1259 - Cloned CLONE_NEWUSER.
      lxc-start p3 20180401142858.633 INFO     lxc_start - start.c:lxc_spawn:1259 - Cloned CLONE_NEWNS.
      lxc-start p3 20180401142858.633 INFO     lxc_start - start.c:lxc_spawn:1259 - Cloned CLONE_NEWPID.
      lxc-start p3 20180401142858.633 INFO     lxc_start - start.c:lxc_spawn:1259 - Cloned CLONE_NEWUTS.
      lxc-start p3 20180401142858.633 INFO     lxc_start - start.c:lxc_spawn:1259 - Cloned CLONE_NEWIPC.
      lxc-start p3 20180401142858.633 DEBUG    lxc_conf - conf.c:idmaptool_on_path_and_privileged:2595 - The binary "/usr/bin/newuidmap" does have the setuid bit set.
      lxc-start p3 20180401142858.633 DEBUG    lxc_conf - conf.c:idmaptool_on_path_and_privileged:2595 - The binary "/usr/bin/newgidmap" does have the setuid bit set.
      lxc-start p3 20180401142858.633 DEBUG    lxc_conf - conf.c:lxc_map_ids:2683 - Functional newuidmap and newgidmap binary found.
      lxc-start p3 20180401142858.636 TRACE    lxc_conf - conf.c:lxc_map_ids:2739 - newuidmap wrote mapping "newuidmap 2184 0 100000 65536"
      lxc-start p3 20180401142858.638 TRACE    lxc_conf - conf.c:lxc_map_ids:2739 - newgidmap wrote mapping "newgidmap 2184 0 100000 65536"
      lxc-start p3 20180401142858.639 INFO     lxc_start - start.c:do_start:848 - Unshared CLONE_NEWNET.
      lxc-start p3 20180401142858.642 TRACE    lxc_conf - conf.c:userns_exec_1:3816 - establishing uid mapping for "2189" in new user namespace: nsuid 0 - hostid 100000 - range 65536
      lxc-start p3 20180401142858.642 TRACE    lxc_conf - conf.c:userns_exec_1:3816 - establishing uid mapping for "2189" in new user namespace: nsuid 65536 - hostid 1000 - range 1
      lxc-start p3 20180401142858.642 TRACE    lxc_conf - conf.c:userns_exec_1:3816 - establishing gid mapping for "2189" in new user namespace: nsuid 0 - hostid 100000 - range 65536
      lxc-start p3 20180401142858.642 TRACE    lxc_conf - conf.c:userns_exec_1:3816 - establishing gid mapping for "2189" in new user namespace: nsuid 65536 - hostid 1000 - range 1
      lxc-start p3 20180401142858.642 DEBUG    lxc_conf - conf.c:idmaptool_on_path_and_privileged:2595 - The binary "/usr/bin/newuidmap" does have the setuid bit set.
      lxc-start p3 20180401142858.642 DEBUG    lxc_conf - conf.c:idmaptool_on_path_and_privileged:2595 - The binary "/usr/bin/newgidmap" does have the setuid bit set.
      lxc-start p3 20180401142858.642 DEBUG    lxc_conf - conf.c:lxc_map_ids:2683 - Functional newuidmap and newgidmap binary found.
   ```   
 - [X] the containers configuration file
   ```
   # Distribution configuration
   lxc.include = /usr/share/lxc/config/ubuntu.common.conf
   lxc.include = /usr/share/lxc/config/ubuntu.userns.conf
   lxc.arch = x86_64

   # Container specific configuration
   lxc.idmap = u 0 100000 65536
   lxc.idmap = g 0 100000 65536
   lxc.rootfs.path = dir:/home/asavonic/.local/share/lxc/p3/rootfs
   lxc.uts.name = p3
   ```
 - `cat /etc/subgid /etc/subuid`
    ```
    asavonic:100000:65537
    asavonic:100000:65537
    ```

**Comment by brauner** [comment]
Can you try and reproduce this with `LXC 3.0` if Void has already updated.

**Comment by asavonic** [comment]
Thanks @brauner, the issue is fixed in 3.0.
